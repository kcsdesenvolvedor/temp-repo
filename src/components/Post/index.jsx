import { React } from 'react';
import P from 'prop-types';
import { PostCard } from '../PostCard';
import './style.css';

export const Post = ({ posts = [] }) => (
  <div className="posts">
    {posts.map((post) => (
      <PostCard key={post.id} id={post.id} title={post.title} body={post.body} cover={post.cover} />
    ))}
  </div>
);

Post.defaultProps = {
  posts: [],
};

Post.propTypes = {
  posts: P.array,
};
