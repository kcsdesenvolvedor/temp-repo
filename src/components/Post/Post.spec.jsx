import { React } from 'react';
import { render, screen } from '@testing-library/react';
import { Post } from '../Post/index';

const props = {
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'body 1',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: 'title 2',
      body: 'body 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body 3',
      cover: 'img/img3.png',
    },
  ],
};
describe('<Post />', () => {
  it('should reder Post correctly', () => {
    render(<Post {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

    expect(screen.getAllByText(/title/i)).toHaveLength(3);

    expect(screen.getAllByText(/body/i)).toHaveLength(3);

    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('should not render Post', () => {
    render(<Post />);

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Post {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
