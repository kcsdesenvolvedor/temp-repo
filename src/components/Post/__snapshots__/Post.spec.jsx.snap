// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img1.png"
    />
    <div
      class="-content"
    >
      <h2>
        title 1
         
        1
      </h2>
      <p>
        body 1
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="img/img2.png"
    />
    <div
      class="-content"
    >
      <h2>
        title 2
         
        2
      </h2>
      <p>
        body 2
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="img/img3.png"
    />
    <div
      class="-content"
    >
      <h2>
        title 3
         
        3
      </h2>
      <p>
        body 3
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
</div>
`;
